src/librustc_lint/types.rs

```diff
@@ -871,7 +868,10 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
 
             ty::Foreign(..) => FfiSafe,
 
-            ty::Param(..) |
+            // `extern "C" fn` functions can have type parameters, which are not FFI-safe,
+            // but they are currently ignored for the purposes of this lint, see #65134.
+            ty::Param(..) | ty::Projection(..) => FfiSafe,
```

Comment:
I'd rather say that we don't know whether the types passed as the parameters are FFI-safe or not (they very well could always be), and we would rather not warn in that case.

Reply:
I've updated the comment here.